sklearn/compose/_column_transformer.py

```diff
@@ -470,6 +488,8 @@ def fit_transform(self, X, y=None):
             sparse matrices.
 
         """
+        # TODO: this should be `feature_names_in_` when we start having it
+        self._feature_names_in = _feature_names(X)
```

Comment:
I would probably do this inline in this PR though we definitely want a function/method later.